PowerballApi.Tests/PowerballRepositoryTests.cs

```diff
@@ -1,140 +1,271 @@
+
+            var drawing = new PowerballSet
+            {
+                Date = DateTime.Parse(dateAfter),
+                PowerPlay = null,
+                WinNumbers = new[] { 1, 2, 3, 4, 5, 6 }
+            };
+            _cacher.Get(Arg.Any<string>()).Returns(new List<PowerballSet>
+
+            {
+                drawing
+            });
+
+            try
+            {
+                _sut.GetByRange(dateAfter, dateBefore);
+            }
+            catch (Exception ex)
+            {
+
+                Assert.IsInstanceOfType(ex, typeof(Exception));
+            }
+        }
+
+        [TestMethod]
+        public void GetByRange_ValidDates_ReturnsDrawingsInRange()
+        {
+            const string dateAfter = "01/01/2017";
+            const string dateBefore = "01/31/2017";
+
+            var drawingList = new List<PowerballSet>
+            {
+                new PowerballSet
+                {
+                    Date = DateTime.Parse("01/01/2017"),
+                    PowerPlay = null,
+                    WinNumbers = new[] {1, 2, 3, 4, 5, 6}
+                },
+                new PowerballSet
+                {
+                    Date = DateTime.Parse("01/10/2017"),
+                    PowerPlay = null,
+                    WinNumbers = new[] {1, 2, 3, 4, 5, 6}
+
+                },
+                new PowerballSet
+                {
+                    Date = DateTime.Parse("01/21/2017"),
+                    PowerPlay = null,
+                    WinNumbers = new[] {1, 2, 3, 4, 5, 6}
+
+                }
+            };
+            _cacher.Get(Arg.Any<string>()).Returns(drawingList);
+
+            var result = _sut.GetByRange(dateAfter, dateBefore);
+
+            Assert.AreEqual(drawingList.Count, result.Count);
+        }
+
+        [TestMethod]
+        public void GetByRange_ValidDates_DoesNotReturnDrawingsOutOfRange()
+        {
+            const string dateAfter = "01/01/2017";
+            const string dateBefore = "01/31/2017";
+
+            var drawingList = new List<PowerballSet>
+            {
+                new PowerballSet
+                {
+                    Date = DateTime.Parse("12/01/2016"),
+                    PowerPlay = null,
+                    WinNumbers = new[] {1, 2, 3, 4, 5, 6}
+                },
+                new PowerballSet
+                {
+                    Date = DateTime.Parse("02/10/2017"),
+                    PowerPlay = null,
+                    WinNumbers = new[] {1, 2, 3, 4, 5, 6}
+
+                },
+                new PowerballSet
+                {
+                    Date = DateTime.Parse("02/21/2017"),
+                    PowerPlay = null,
+                    WinNumbers = new[] {1, 2, 3, 4, 5, 6}
+
+                }
+            };
+            _cacher.Get(Arg.Any<string>()).Returns(drawingList);
+
+            var result = _sut.GetByRange(dateAfter, dateBefore);
+
+            Assert.AreEqual(0, result.Count);
+
+        }
+    }
```

Comment:
Please add a test that hits the final exception. This can be done by having both the substitute for the Cacher and the substitute for the Parser return null. This is a more complex test, but good practice.